x-pack/plugins/infra/public/pages/logs/log_entry_rate/page_results_content.tsx

```diff
@@ -79,6 +84,30 @@ export const LogEntryRateResultsContent: React.FunctionComponent = () => {
     lastChangedTime: Date.now(),
   }));
 
+  const history = useHistory();
+  const linkToLogStream = useCallback<LogEntryFlyoutProps['setFilter']>(
```

Comment:
I had a reason to not do it... but I don't remember it now so I'll change it

Reply:
ah, the moment I tried to redo it I saw why I did it this way.

The link is constructed with the `filter`, `id` and `timeKey` parameters, which come from the `<LogEntryFlyout setFilter />` callback. `useLinkProps` cannot be called inside a callback, because hooks.

Something I can do to remove the usage of `history` is use `navigateToApp` as it's used [inside the `useLinkProps` hook](https://github.com/elastic/kibana/blob/89dba3927333fc2f6b60573a1e173d966510ee41/x-pack/plugins/infra/public/hooks/use_link_props.tsx#L79). What do you think?

Reply:
Ah, okay. Yeah `navigateToApp` would be better, thanks 👍